<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Pre-processing Sequence Reads | High Protein Diet Microbiome Analysis</title>
  <meta name="description" content="This is the workflow for 16S analysis for the high protein diet study" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Pre-processing Sequence Reads | High Protein Diet Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is the workflow for 16S analysis for the high protein diet study" />
  <meta name="github-repo" content="mattsnelson/hpd_16S" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Pre-processing Sequence Reads | High Protein Diet Microbiome Analysis" />
  
  <meta name="twitter:description" content="This is the workflow for 16S analysis for the high protein diet study" />
  

<meta name="author" content="Matthew Snelson" />


<meta name="date" content="2020-07-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="setup-and-qc.html"/>
<link rel="next" href="taxonomic-analysis.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">High Protein Diet study - Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a></li>
<li class="chapter" data-level="2" data-path="setup-and-qc.html"><a href="setup-and-qc.html"><i class="fa fa-check"></i><b>2</b> Setup and QC</a><ul>
<li class="chapter" data-level="2.1" data-path="setup-and-qc.html"><a href="setup-and-qc.html#install-qiime-and-acitvate-conda-environment"><i class="fa fa-check"></i><b>2.1</b> Install QIIME and acitvate conda environment</a></li>
<li class="chapter" data-level="2.2" data-path="setup-and-qc.html"><a href="setup-and-qc.html#quality-check-fastqc"><i class="fa fa-check"></i><b>2.2</b> Quality Check (FastQC)</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html"><i class="fa fa-check"></i><b>3</b> Pre-processing Sequence Reads</a><ul>
<li class="chapter" data-level="3.1" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#import-data"><i class="fa fa-check"></i><b>3.1</b> Import data</a></li>
<li class="chapter" data-level="3.2" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#denoise-with-dada2"><i class="fa fa-check"></i><b>3.2</b> Denoise with dada2</a></li>
<li class="chapter" data-level="3.3" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#check-denoising-stats"><i class="fa fa-check"></i><b>3.3</b> Check denoising stats</a></li>
<li class="chapter" data-level="3.4" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#get-featuretable-frequency-and-featuredata-sequences"><i class="fa fa-check"></i><b>3.4</b> Get FeatureTable [frequency] and FeatureData [sequences]</a></li>
<li class="chapter" data-level="3.5" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#train-classifier-for-doing-taxonomy"><i class="fa fa-check"></i><b>3.5</b> Train Classifier (for doing taxonomy)</a></li>
<li class="chapter" data-level="3.6" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#classify-rep-seqs"><i class="fa fa-check"></i><b>3.6</b> Classify rep seqs</a></li>
<li class="chapter" data-level="3.7" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#create-a-phylogenetic-tree"><i class="fa fa-check"></i><b>3.7</b> Create a phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="taxonomic-analysis.html"><a href="taxonomic-analysis.html"><i class="fa fa-check"></i><b>4</b> Taxonomic Analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="taxonomic-analysis.html"><a href="taxonomic-analysis.html#taxonomic-analysis---taxa-barplots"><i class="fa fa-check"></i><b>4.1</b> Taxonomic analysis - taxa barplots</a></li>
<li class="chapter" data-level="4.2" data-path="taxonomic-analysis.html"><a href="taxonomic-analysis.html#generate-heatmap"><i class="fa fa-check"></i><b>4.2</b> Generate heatmap</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="alpha-and-beta-diversity-analyses.html"><a href="alpha-and-beta-diversity-analyses.html"><i class="fa fa-check"></i><b>5</b> Alpha and Beta Diversity Analyses</a><ul>
<li class="chapter" data-level="5.1" data-path="alpha-and-beta-diversity-analyses.html"><a href="alpha-and-beta-diversity-analyses.html#alpha-rarefaction-curves"><i class="fa fa-check"></i><b>5.1</b> Alpha Rarefaction Curves</a></li>
<li class="chapter" data-level="5.2" data-path="alpha-and-beta-diversity-analyses.html"><a href="alpha-and-beta-diversity-analyses.html#diversity-metrics"><i class="fa fa-check"></i><b>5.2</b> Diversity Metrics</a></li>
<li class="chapter" data-level="5.3" data-path="alpha-and-beta-diversity-analyses.html"><a href="alpha-and-beta-diversity-analyses.html#beta-signficance-metrics"><i class="fa fa-check"></i><b>5.3</b> Beta signficance metrics</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="picrust2.html"><a href="picrust2.html"><i class="fa fa-check"></i><b>6</b> Picrust2</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">High Protein Diet Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pre-processing-sequence-reads" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Pre-processing Sequence Reads</h1>
<div id="import-data" class="section level2">
<h2><span class="header-section-number">3.1</span> Import data</h2>
<p>First up gotta import reads into a qiime artifact (called here <code>input-data-casava</code>)</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb7-1" data-line-number="1">$ <span class="ex">qiime</span> tools import \</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">--type <span class="st">&#39;SampleData[PairedEndSequencesWithQuality]&#39;</span> \</a>
<a class="sourceLine" id="cb7-3" data-line-number="3">--input-path data-raw/ \</a>
<a class="sourceLine" id="cb7-4" data-line-number="4">--input-format CasavaOneEightSingleLanePerSampleDirFmt \</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">--output-path input-data-casava.qza</a></code></pre></div>
</div>
<div id="denoise-with-dada2" class="section level2">
<h2><span class="header-section-number">3.2</span> Denoise with dada2</h2>
<p>This step takes a while. so may want to setup within a tmux session if running on nectar cloud. Note I use a trim-left parameter of 15 on both the forward and reverse reads to deal with the amplicon primers (as seems to work better than using cut-adapt).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb8-1" data-line-number="1">$ <span class="ex">qiime</span> dada2 denoise-paired \</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">--p-n-threads 8 \</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">--i-demultiplexed-seqs input-data-casava.qza \</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">--p-trim-left-f 15 \</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">--p-trim-left-r 15 \</a>
<a class="sourceLine" id="cb8-6" data-line-number="6">--p-trunc-len-f 260 \</a>
<a class="sourceLine" id="cb8-7" data-line-number="7">--p-trunc-len-r 215 \</a>
<a class="sourceLine" id="cb8-8" data-line-number="8">--output-dir DADA2_denoising_output \</a>
<a class="sourceLine" id="cb8-9" data-line-number="9">--verbose \</a>
<a class="sourceLine" id="cb8-10" data-line-number="10"><span class="op">&amp;&gt;</span> DADA2_denoising.log</a></code></pre></div>
<p>So now we have a folder, <code>DADA2_denoisng_output</code> with following items in there:
- denoising_stats.qza
- representative_sequences.qza
- table.qza</p>
</div>
<div id="check-denoising-stats" class="section level2">
<h2><span class="header-section-number">3.3</span> Check denoising stats</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb9-1" data-line-number="1">$ <span class="bu">cd</span> DADA2_denoising_output</a>
<a class="sourceLine" id="cb9-2" data-line-number="2"></a>
<a class="sourceLine" id="cb9-3" data-line-number="3">$ <span class="ex">qiime</span> metadata tabulate \</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">--m-input-file denoising_stats.qza \</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">--o-visualization denoising_stats.qzv \</a></code></pre></div>
</div>
<div id="get-featuretable-frequency-and-featuredata-sequences" class="section level2">
<h2><span class="header-section-number">3.4</span> Get FeatureTable [frequency] and FeatureData [sequences]</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb10-1" data-line-number="1">$ <span class="ex">qiime</span> feature-table summarize \</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">  --i-table table.qza \</a>
<a class="sourceLine" id="cb10-3" data-line-number="3">  --o-visualization table.qzv \</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">  --m-sample-metadata-file ../metadata.txt </a>
<a class="sourceLine" id="cb10-5" data-line-number="5"></a>
<a class="sourceLine" id="cb10-6" data-line-number="6">$ <span class="ex">qiime</span> feature-table tabulate-seqs \</a>
<a class="sourceLine" id="cb10-7" data-line-number="7">  --i-data representative_sequences.qza \</a>
<a class="sourceLine" id="cb10-8" data-line-number="8">  --o-visualization representative_sequences.qzv</a></code></pre></div>
<p>Note: min freq: 20,718.0 reads</p>
</div>
<div id="train-classifier-for-doing-taxonomy" class="section level2">
<h2><span class="header-section-number">3.5</span> Train Classifier (for doing taxonomy)</h2>
<p>First up, we gotta import a the greengenes database stuff into qiime as qiime artifacts.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb11-1" data-line-number="1">$ <span class="fu">mkdir</span> classifier</a>
<a class="sourceLine" id="cb11-2" data-line-number="2">$ <span class="bu">cd</span> classifier</a>
<a class="sourceLine" id="cb11-3" data-line-number="3"></a>
<a class="sourceLine" id="cb11-4" data-line-number="4"><span class="co"># download green genes 13_8</span></a>
<a class="sourceLine" id="cb11-5" data-line-number="5">$ <span class="fu">wget</span> ftp://greengenes.microbio.me/greengenes_release/gg_13_5/gg_13_8_otus.tar.gz</a>
<a class="sourceLine" id="cb11-6" data-line-number="6">$ <span class="fu">gunzip</span> gg_13_8_otus.tar.gz</a>
<a class="sourceLine" id="cb11-7" data-line-number="7">$ <span class="fu">tar</span> -xvf gg_13_8_otus.tar</a>
<a class="sourceLine" id="cb11-8" data-line-number="8">$ <span class="fu">rm</span> gg_13_8_otus.tar</a>
<a class="sourceLine" id="cb11-9" data-line-number="9"></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"> <span class="co"># the sequences from greengenes</span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11">$ <span class="ex">qiime</span> tools import \</a>
<a class="sourceLine" id="cb11-12" data-line-number="12">--type <span class="st">&#39;FeatureData[Sequence]&#39;</span> \</a>
<a class="sourceLine" id="cb11-13" data-line-number="13">--input-path gg_13_8_otus/rep_set/99_otus.fasta \</a>
<a class="sourceLine" id="cb11-14" data-line-number="14">--output-path 99_otus.qza</a>
<a class="sourceLine" id="cb11-15" data-line-number="15"></a>
<a class="sourceLine" id="cb11-16" data-line-number="16"><span class="co"># The taxonomy strings from greengenes</span></a>
<a class="sourceLine" id="cb11-17" data-line-number="17">$ <span class="ex">qiime</span> tools import \</a>
<a class="sourceLine" id="cb11-18" data-line-number="18">--type <span class="st">&#39;FeatureData[Taxonomy]&#39;</span> \</a>
<a class="sourceLine" id="cb11-19" data-line-number="19">--input-format HeaderlessTSVTaxonomyFormat \</a>
<a class="sourceLine" id="cb11-20" data-line-number="20">--input-path gg_13_8_otus/taxonomy/99_otu_taxonomy.txt \</a>
<a class="sourceLine" id="cb11-21" data-line-number="21">--output-path 99_otus_16S_taxonomy.qza</a></code></pre></div>
<p>Extract reads from the relevant region (V3-V5) out of the ref database</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb12-1" data-line-number="1">$ <span class="ex">qiime</span> feature-classifier extract-reads \</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">--i-sequences 99_otus.qza \</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">--p-f-primer CCTACGGGNGGCWGCAG  \</a>
<a class="sourceLine" id="cb12-4" data-line-number="4">--p-r-primer GACTACHVGGGTATCTAATCC   \</a>
<a class="sourceLine" id="cb12-5" data-line-number="5">--p-min-length 300 \</a>
<a class="sourceLine" id="cb12-6" data-line-number="6">--p-max-length 600 \</a>
<a class="sourceLine" id="cb12-7" data-line-number="7">--o-reads ref_seqs.qza \</a>
<a class="sourceLine" id="cb12-8" data-line-number="8">--verbose \</a>
<a class="sourceLine" id="cb12-9" data-line-number="9"><span class="op">&amp;&gt;</span> 16S_training.log</a></code></pre></div>
<p>Now, train the classifier on this region. I.e use this <code>ref_seqs.qza</code> qiime artifact as the input for feature-classifier.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb13-1" data-line-number="1">$ <span class="ex">qiime</span> feature-classifier fit-classifier-naive-bayes \</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">--i-reference-reads ref_seqs.qza \</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">--i-reference-taxonomy 99_otus_16S_taxonomy.qza \</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">--o-classifier classifier_16S.qza \</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">--verbose \</a>
<a class="sourceLine" id="cb13-6" data-line-number="6"><span class="op">&amp;&gt;</span> 16S_classifier.log</a></code></pre></div>
<p>The output of this is our classifier, which I’ve called <code>classifier_16S.qza</code>, is what will be used in the next step.</p>
</div>
<div id="classify-rep-seqs" class="section level2">
<h2><span class="header-section-number">3.6</span> Classify rep seqs</h2>
<p>So now we’ve got the classifier, we can use it to classify our representative seqs (rep-seqs.qza), which I then tabulated and visualised:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb14-1" data-line-number="1">$ <span class="bu">cd</span> ..</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">$ <span class="fu">mkdir</span> taxonomy</a>
<a class="sourceLine" id="cb14-3" data-line-number="3"></a>
<a class="sourceLine" id="cb14-4" data-line-number="4">$ <span class="ex">qiime</span> feature-classifier classify-sklearn \</a>
<a class="sourceLine" id="cb14-5" data-line-number="5">--i-classifier classifier/classifier_16S.qza \</a>
<a class="sourceLine" id="cb14-6" data-line-number="6">--i-reads DADA2_denoising_output/representative_sequences.qza \</a>
<a class="sourceLine" id="cb14-7" data-line-number="7">--o-classification taxonomy/classified_rep_seqs.qza</a>
<a class="sourceLine" id="cb14-8" data-line-number="8"></a>
<a class="sourceLine" id="cb14-9" data-line-number="9"><span class="co"># Tabulate the features, their taxonomy and the confidence of taxonomy assignment</span></a>
<a class="sourceLine" id="cb14-10" data-line-number="10">$ <span class="bu">cd</span> taxonomy</a>
<a class="sourceLine" id="cb14-11" data-line-number="11"></a>
<a class="sourceLine" id="cb14-12" data-line-number="12">$ <span class="ex">qiime</span> metadata tabulate \</a>
<a class="sourceLine" id="cb14-13" data-line-number="13">--m-input-file classified_rep_seqs.qza \</a>
<a class="sourceLine" id="cb14-14" data-line-number="14">--o-visualization classified_rep_seqs.qzv</a></code></pre></div>
</div>
<div id="create-a-phylogenetic-tree" class="section level2">
<h2><span class="header-section-number">3.7</span> Create a phylogenetic tree</h2>
<p>Note that this only needs the rep seqs (not the classified rep seqs)</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb15-1" data-line-number="1">$ <span class="bu">cd</span> ..</a>
<a class="sourceLine" id="cb15-2" data-line-number="2"></a>
<a class="sourceLine" id="cb15-3" data-line-number="3">$ <span class="ex">qiime</span> phylogeny align-to-tree-mafft-fasttree \</a>
<a class="sourceLine" id="cb15-4" data-line-number="4">--i-sequences DADA2_denoising_output/representative_sequences.qza \</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">--output-dir phylogenetic_tree \</a>
<a class="sourceLine" id="cb15-6" data-line-number="6">--p-n-threads 8 \</a>
<a class="sourceLine" id="cb15-7" data-line-number="7">--verbose \</a>
<a class="sourceLine" id="cb15-8" data-line-number="8"><span class="op">&amp;&gt;</span> phylogenetic_tree_generation.log</a></code></pre></div>
<p>Note: it’s the rooted_tree.qza file which will be used in downstream analyses.</p>
<p>Now on to the fun stuff - taxonomy and diversity analyses!</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="setup-and-qc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="taxonomic-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-preprocess.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["hpd-microbiome.pdf", "hpd-microbiome.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
