<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Setup and QC | High Protein Diet Microbiome Analysis</title>
  <meta name="description" content="This is the workflow for 16S analysis for the high protein diet study" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Setup and QC | High Protein Diet Microbiome Analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is the workflow for 16S analysis for the high protein diet study" />
  <meta name="github-repo" content="mattsnelson/hpd_16S" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Setup and QC | High Protein Diet Microbiome Analysis" />
  
  <meta name="twitter:description" content="This is the workflow for 16S analysis for the high protein diet study" />
  

<meta name="author" content="Matthew Snelson" />


<meta name="date" content="2022-02-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="pre-processing-sequence-reads.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">High Protein Diet study - Microbiome Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a></li>
<li class="chapter" data-level="2" data-path="setup-and-qc.html"><a href="setup-and-qc.html"><i class="fa fa-check"></i><b>2</b> Setup and QC</a><ul>
<li class="chapter" data-level="2.1" data-path="setup-and-qc.html"><a href="setup-and-qc.html#install-qiime-and-acitvate-conda-environment"><i class="fa fa-check"></i><b>2.1</b> Install QIIME and acitvate conda environment</a></li>
<li class="chapter" data-level="2.2" data-path="setup-and-qc.html"><a href="setup-and-qc.html#quality-check-fastqc"><i class="fa fa-check"></i><b>2.2</b> Quality Check (FastQC)</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html"><i class="fa fa-check"></i><b>3</b> Pre-processing Sequence Reads</a><ul>
<li class="chapter" data-level="3.1" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#import-data"><i class="fa fa-check"></i><b>3.1</b> Import data</a></li>
<li class="chapter" data-level="3.2" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#denoise-with-dada2"><i class="fa fa-check"></i><b>3.2</b> Denoise with dada2</a></li>
<li class="chapter" data-level="3.3" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#check-denoising-stats"><i class="fa fa-check"></i><b>3.3</b> Check denoising stats</a></li>
<li class="chapter" data-level="3.4" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#get-featuretable-frequency-and-featuredata-sequences"><i class="fa fa-check"></i><b>3.4</b> Get FeatureTable [frequency] and FeatureData [sequences]</a></li>
<li class="chapter" data-level="3.5" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#train-classifier-for-doing-taxonomy"><i class="fa fa-check"></i><b>3.5</b> Train Classifier (for doing taxonomy)</a></li>
<li class="chapter" data-level="3.6" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#classify-rep-seqs"><i class="fa fa-check"></i><b>3.6</b> Classify rep seqs</a></li>
<li class="chapter" data-level="3.7" data-path="pre-processing-sequence-reads.html"><a href="pre-processing-sequence-reads.html#create-a-phylogenetic-tree"><i class="fa fa-check"></i><b>3.7</b> Create a phylogenetic tree</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="taxonomic-analysis.html"><a href="taxonomic-analysis.html"><i class="fa fa-check"></i><b>4</b> Taxonomic Analysis</a><ul>
<li class="chapter" data-level="4.1" data-path="taxonomic-analysis.html"><a href="taxonomic-analysis.html#taxonomic-analysis---taxa-barplots"><i class="fa fa-check"></i><b>4.1</b> Taxonomic analysis - taxa barplots</a></li>
<li class="chapter" data-level="4.2" data-path="taxonomic-analysis.html"><a href="taxonomic-analysis.html#generate-heatmap"><i class="fa fa-check"></i><b>4.2</b> Generate heatmap</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="alpha-and-beta-diversity-analyses.html"><a href="alpha-and-beta-diversity-analyses.html"><i class="fa fa-check"></i><b>5</b> Alpha and Beta Diversity Analyses</a><ul>
<li class="chapter" data-level="5.1" data-path="alpha-and-beta-diversity-analyses.html"><a href="alpha-and-beta-diversity-analyses.html#alpha-rarefaction-curves"><i class="fa fa-check"></i><b>5.1</b> Alpha Rarefaction Curves</a></li>
<li class="chapter" data-level="5.2" data-path="alpha-and-beta-diversity-analyses.html"><a href="alpha-and-beta-diversity-analyses.html#diversity-metrics"><i class="fa fa-check"></i><b>5.2</b> Diversity Metrics</a></li>
<li class="chapter" data-level="5.3" data-path="alpha-and-beta-diversity-analyses.html"><a href="alpha-and-beta-diversity-analyses.html#beta-signficance-metrics"><i class="fa fa-check"></i><b>5.3</b> Beta signficance metrics</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="picrust2.html"><a href="picrust2.html"><i class="fa fa-check"></i><b>6</b> Picrust2</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">High Protein Diet Microbiome Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="setup-and-qc" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> Setup and QC</h1>
<div id="install-qiime-and-acitvate-conda-environment" class="section level2">
<h2><span class="header-section-number">2.1</span> Install QIIME and acitvate conda environment</h2>
<p>This anlaysis was coducted using QIIME 2020.2. First step, download the yml file and then use it to create a conda environment for the install.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="setup-and-qc.html#cb1-1"></a>$ <span class="fu">wget</span> https://data.qiime2.org/distro/core/qiime2-2020.2-py36-linux-conda.yml</span>
<span id="cb1-2"><a href="setup-and-qc.html#cb1-2"></a>$ <span class="ex">conda</span> env create -n qiime2-2020.2 --file qiime2-2020.2-py36-linux-conda.yml</span></code></pre></div>
<p>Then activate the conda environment</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="setup-and-qc.html#cb2-1"></a>$ <span class="ex">conda</span> activate qiime2-2020.2</span></code></pre></div>
</div>
<div id="quality-check-fastqc" class="section level2">
<h2><span class="header-section-number">2.2</span> Quality Check (FastQC)</h2>
<p>Run fastqc. Note that these are gzipped fastqc files, but fastqc still works.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="setup-and-qc.html#cb3-1"></a>$ <span class="ex">fastqc</span> data-raw/*.fastq.gz</span></code></pre></div>
<p>Running fastqc generates a fastqc.zip file and fastqc.html per fastq file. Move the fastqc generated files into a <code>results/fastqc</code> folder.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="setup-and-qc.html#cb4-1"></a>$ <span class="fu">mkdir</span> results/</span>
<span id="cb4-2"><a href="setup-and-qc.html#cb4-2"></a>$ <span class="fu">mkdir</span> results/fastqc</span>
<span id="cb4-3"><a href="setup-and-qc.html#cb4-3"></a>$ <span class="fu">mv</span> data-raw/*fastqc* results/fastqc/</span></code></pre></div>
<p>It’s also handy to run mutliQC as this allows to look at quality of all samples at once (rather than one at a time with fastQC reports)</p>
<p>First up, install multiQC (if not already installed)</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="setup-and-qc.html#cb5-1"></a>$ <span class="ex">conda</span> install -c bioconda -c conda-forge multiqc</span></code></pre></div>
<p>Then run multiqc on the <code>results/fastqc</code> folder.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="setup-and-qc.html#cb6-1"></a>$ <span class="fu">mkdir</span> results/multiqc</span>
<span id="cb6-2"><a href="setup-and-qc.html#cb6-2"></a>$ <span class="ex">multiqc</span> results/fastqc/* -o results/multiqc/</span></code></pre></div>
<p>Note: FastQC generates fastqc reports for individual files, the benefit of using mutliqc is that we get a report where we can see all the samples at once. Benefit here is if there’s one real stinker you’ll be able to see it which you might miss if looking at individual fastqc reports. In the plot below the red are all the reverse reads, the quality of the reverse reads tends to always drop off sooner than the forward reads.</p>
<div class="figure">
<img src="figs/fastqc_per_base_sequence_quality_plot.png" alt="" />
<p class="caption">multiqc results</p>
</div>
<p>Based on the multiQC report: will choose limits of F 260 bp and R : 215 bp</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pre-processing-sequence-reads.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/01-intro.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["hpd-microbiome.pdf", "hpd-microbiome.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
